src/software/firmware/srcs/integration_test.cpp

```diff
@@ -0,0 +1,253 @@
+/******************************************************************************
+ * @author Makers For Life
+ * @copyright Copyright (c) 2020 Makers For Life
+ * @file integration_test.cpp
+ * @brief Entry point of integration test program
+ *****************************************************************************/
+
+#pragma once
+
+#include "../includes/config.h"
+#if MODE == MODE_INTEGRATION_TEST
+
+// INCLUDES ===================================================================
+
+// External
+#include <Arduino.h>
+#include <OneButton.h>
+
+// Internal
+#include "../includes/battery.h"
+#include "../includes/blower.h"
+#include "../includes/debug.h"
+#include "../includes/parameters.h"
+#include "../includes/pression.h"
+#include "../includes/pressure_valve.h"
+#include "../includes/screen.h"
+
+// State machine states
+#define STEP_WELCOME 0
+#define STEP_BLOWER_TEST 1
+#define STEP_VALVE_BLOWER_TEST 2
+#define STEP_VALVE_PATIENT_TEST 3
+#define STEP_VALVE_BLOWER_LEAK_TEST 4
+#define STEP_VALVE_PATIENT_LEAK_TEST 5
+#define STEP_PRESSURE_TEST 6
+#define STEP_BATTERY_TEST 7
+
+#define NUMBER_OF_STATES 8
+
+#define UNGREEDY(is_drawn, statement)                                                              \
+    if (is_drawn == 0) {                                                                           \
+        statement;                                                                                 \
+        is_drawn = 1;                                                                              \
+    }
+
```

Comment:
Should we merge this macro between `integration_test.cpp` and `qualification.cpp`?